modules/tinymce/src/plugins/lists/test/ts/browser/IndentTest.ts

```diff
@@ -522,6 +522,45 @@ describe('browser.tinymce.plugins.lists.IndentTest', () => {
     assert.equal(editor.selection.getNode().nodeName, 'LI');
   });
 
+  it('TINY-10414: toggle a list inside a `li` with non-list siblings should not delete the siblings', () => {
```

Comment:
I wonder if we should use property based tests here. There is a ListModeTest that uses fast check. In theory one could generate a list with UL/OL/LI items with N paragraphs before/after the UL/OL then run indent on any of the LI should never remove the paragraphs. So one could generate a structure, apply an operation and the check if the non LI elements are still the same. This way we would find any edge case that we are not covering here.